substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/PointsToAnalyzer.java

```diff
@@ -0,0 +1,395 @@
+    private PointsToAnalyzer(OptionValues options) {
+        this.options = options;
+        analysisFeatureManager = new AnalysisFeatureManager(options);
+        String appCP = PointstoOptions.AnalysisTargetAppCP.getValue(options);
+        if (appCP == null) {
+            AnalysisError.shouldNotReachHere("Must specify analysis target application's classpath with -H:" + PointstoOptions.AnalysisTargetAppCP.getName());
+        }
+        List<URL> urls = new ArrayList<>();
+        for (String cp : appCP.split(File.pathSeparator)) {
+            try {
+                File file = new File(cp);
+                if (file.exists()) {
+                    urls.add(file.toURI().toURL());
+                }
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            }
+        }
+
+        analysisClassLoader = new AnalysisClassLoader(urls.toArray(new URL[0]), ClassLoader.getPlatformClassLoader());
+        Providers providers = getGraalCapability(RuntimeProvider.class).getHostBackend().getProviders();
+        SnippetReflectionProvider snippetReflection = getGraalCapability(SnippetReflectionProvider.class);
+        MetaAccessProvider originalMetaAccess = providers.getMetaAccess();
+        debugContext = new DebugContext.Builder(options, new GraalDebugHandlersFactory(snippetReflection)).build();
+        ForkJoinPool executor = PointsToAnalysis.createExecutor(debugContext, Math.min(Runtime.getRuntime().availableProcessors(), 32));
+        HotSpotHost hotSpotHost = new HotSpotHost(options, analysisClassLoader);
+        int wordSize = getWordSize();
+        AnalysisPolicy analysisPolicy = PointstoOptions.AllocationSiteSensitiveHeap.getValue(options) ? new BytecodeSensitiveAnalysisPolicy(options)
+                        : new DefaultAnalysisPolicy(options);
+
+        JavaKind wordKind = JavaKind.fromWordSize(wordSize);
+        AccessControllerSubstitutionProcessor accessControllerSubstitutionProcessor = new AccessControllerSubstitutionProcessor(originalMetaAccess, analysisClassLoader);
+        AnalysisUniverse aUniverse = new AnalysisUniverse(hotSpotHost, wordKind, analysisPolicy,
+                        SubstitutionProcessor.chain(accessControllerSubstitutionProcessor, SubstitutionProcessor.IDENTITY),
+                        originalMetaAccess, snippetReflection, snippetReflection, new PointsToAnalysisFactory());
+        AnalysisMetaAccess aMetaAccess = new AnalysisMetaAccess(aUniverse, originalMetaAccess);
+        aMetaAccess.lookupJavaType(String.class).registerAsReachable();
+        PointstoConstantReflectionProvider aConstantReflection = new PointstoConstantReflectionProvider(aUniverse, HotSpotJVMCIRuntime.runtime());
+        PointstoConstantFieldProvider aConstantFieldProvider = new PointstoConstantFieldProvider(aMetaAccess);
+        WordTypes aWordTypes = new WordTypes(aMetaAccess, wordKind);
+        PointstoStampProvider aStampProvider = new PointstoStampProvider(aMetaAccess);
```

Comment:
Why do you need a custom stamp provider? You could get the original Graal providers from `Providers originalProviders = GraalAccess.getOriginalProviders();`.